cai_causal_graph/causal_graph.py

```diff
@@ -695,10 +773,7 @@ def add_node(
                 'or provide a constructed `Node` object using the `node` parameter.'
             )
 
-        identifier = Node.identifier_from(identifier)
-
-        if identifier in self._nodes_by_identifier:
-            raise CausalGraphErrors.NodeDuplicatedError(f'Node already exists: {identifier}')
+        identifier = self._check_node_exists(identifier)
 
         node = Node(identifier, variable_type=variable_type)
```

Comment:
Another spot that should be `_NodeCls(identifier, variable_type=variable_type)`. Will stop commenting as it needs to be changed throughout.